addons/test_mail/tests/test_mail_template_preview.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.test_mail.tests.common import TestMailCommon, TestRecipients
+from odoo.tests import tagged
+
+
+@tagged('mail_template')
+class TestMailTemplateTools(TestMailCommon, TestRecipients):
+    @classmethod
+    def setUpClass(cls):
+        super(TestMailTemplateTools, cls).setUpClass()
+        cls.test_record = cls.env['mail.test.simple'].with_context(cls._test_context).create({
+            'email_from': 'ignasse@example.com',
+            'name': 'Test',
+        })
+
+        cls.test_template = cls._create_template('mail.test.simple', {
+            'body_html': '<p>EnglishBody for <t t-out="object.name"/></p>',
+            'email_to': '%s, %s' % ('test1@example.com', 'test2@example.com'),
+            'email_cc': '%s' % cls.partner_1.email,
+            'partner_to': '%s,%s' % (cls.partner_2.id, cls.user_admin.partner_id.id),
+            'subject': 'EnglishSubject for {{ object.name }}',
+        })
+
+    def test_initial_values(self):
+        self.assertTrue(self.test_template.email_to)
+        self.assertTrue(self.test_template.email_cc)
+        self.assertEqual(len(self.test_template.partner_to.split(',')), 2)
+        self.assertTrue(self.test_record.email_from)
+
+    def test_mail_template_preview_recipients(self):
+        preview = self.env['mail.template.preview'].create({
```

Comment:
We could probably use the 'Form' tool to simulate the form view, that way it tests a bit more a 'real life' use case :)

Reply:
Yes indeed, thanks, it’s done. I have also splitted the fix and the test in 2 commits as you have suggested me in another PR (in case the fix is reverted).